pkg/service/catalog.go

```diff
@@ -271,6 +299,28 @@ func NewPlanFromJSON(jsonBytes []byte) (Plan, error) {
 	if err := json.Unmarshal(jsonBytes, p); err != nil {
 		return nil, err
 	}
+	if p.ParameterSchemas.ServiceInstances != nil {
```

Comment:
the following lines are for loading the catalog from json and repopulating the PlanProperties correctly.

Reply:
I'm wondering if it's really necessary to straighten out the properties. The original intention was that properties would only be used as input to `NewPlan(...)`. After that, they shouldn't matter...

I realize, as I type this, that the fact that a `plan` is _composed_ of `PlanProperties` makes the statement above less obvious, or perhaps not entirely true.

Two things:

1. I'd like to see if we can reasonably mitigate the "wrong" composition about in a separate PR. I have some idea of how it can be done, but it might not pan out.

2. The `NewPlanFromJSON (...)` function will be dead code after #357 is addressed.

Bottom line is I don't think we need this code here, but there's also no harm in it staying for now.